src/callbacks.py

```diff
@@ -132,7 +132,7 @@ def update_components(n_clicks_next, n_clicks_back, config, n_clicks_sdgs, comme
         session_ids[idx_next] = doc['_id']
```

Comment:
```suggestion
  if idx_next == len(session_ids):
        final_layout = components.get_finish_layout(reason='session_done')
        return no_update, no_update, no_update, no_update, final_layout, config, no_update

    doc_id = session_ids[idx_next]
    if doc_id is None:
        doc = database.get_paragraph(language, email)
        selected_sgds, comment = None, None
    else:
        doc = database.get_paragraph_by_id(doc_id)
        annotation = utils.get_user_annotation(doc, email)
        selected_sgds, comment = annotation['labels'], annotation['comment']

    if doc is None:
        final_layout = components.get_finish_layout(reason='no_tasks')
        return no_update, no_update, no_update, no_update, final_layout, config, no_update
    session_ids[idx_next] = doc['_id']
```

Reply:
`if idx_next == len(session_ids)` should be checked before `doc_id = session_ids[idx_next]`, otherwise this leads to IndexError when last text is reached

Reply:
This is already addressed in the latest commit